### PR TITLE
Fix: fallback to _parse_term when parsing INTERVAL values

### LINKED ISSUE #1478: SQL with INTERVAL key word parse error
```python
from sqlglot import parse_one
sql = """
SELECT DATE(DATE_SUB(`dt`, INTERVAL DAYOFMONTH(`dt`) - 1 DAY)) AS __timestamp
from tableT
"""
expression_tree = parse_one(sql, read="mysql")
print(expression_tree.sql(dialect="mysql", pretty=True))
# Got wrong sql
# SELECT
#   DATE(DATE_SUB(`dt`, INTERVAL (INTERVAL (DAYOFMONTH(`dt`)) - 1) DAY)) AS __timestamp
# FROM tableT

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Fixes #1478

### CODE REVIEW

File: tests/dialects/test_mysql.py [hunk 1]
```diff
@@ -317,6 +317,12 @@ def test_mysql_time(self):
         self.validate_identity("TIME_STR_TO_UNIX(x)", "UNIX_TIMESTAMP(x)")
 
     def test_mysql(self):
+        self.validate_all(
+            "SELECT DATE(DATE_SUB(`dt`, INTERVAL DAYOFMONTH(`dt`) - 1 DAY)) AS __timestamp FROM tableT",
+            write={
+                "mysql": "SELECT DATE(DATE_SUB(`dt`, INTERVAL (DAYOFMONTH(`dt`) - 1) DAY)) AS __timestamp FROM tableT",
```
Comment (GeorgeSittas) [inline tests/dialects/test_mysql.py:323]: FYI: parentheses don't break this query:
```sql
mysql> SELECT DATE_SUB('2020-01-01', INTERVAL (DAYOFMONTH('2020-01-05') - 5) DAY);
+---------------------------------------------------------------------+
| DATE_SUB('2020-01-01', INTERVAL (DAYOFMONTH('2020-01-05') - 5) DAY) |
+---------------------------------------------------------------------+
| 2020-01-01                                                          |
+---------------------------------------------------------------------+
1 row in set (0.00 sec)
```

**Comment by barakalon** [comment]
This is also breaking our stuff.

```python
parse_one("SELECT * FROM x WHERE a <= DATE_ADD(b, INTERVAL -1 YEAR)", read="mysql").sql(dialect="mysql")
# SELECT * FROM x WHERE a <= DATE_ADD(b, INTERVAL (-1) YEAR)
```

I guess the parentheses are ok?

**Comment by GeorgeSittas** [comment]
> ```python
> DATE_ADD(b, INTERVAL -1 YEAR)
> ```

Yep, seems like it:
```sql
mysql> SELECT DATE_ADD('2020-01-01', INTERVAL -1 YEAR);
+------------------------------------------+
| DATE_ADD('2020-01-01', INTERVAL -1 YEAR) |
+------------------------------------------+
| 2019-01-01                               |
+------------------------------------------+
1 row in set (0.00 sec)

mysql> SELECT DATE_ADD('2020-01-01', INTERVAL (-1) YEAR);
+--------------------------------------------+
| DATE_ADD('2020-01-01', INTERVAL (-1) YEAR) |
+--------------------------------------------+
| 2019-01-01                                 |
+--------------------------------------------+
1 row in set (0.00 sec)
```

**Comment by barakalon** [comment]
Nope, all looks good

**Pull request merged**